geoips/interfaces/text_based/ascii_palettes.py

```diff
@@ -0,0 +1,200 @@
+    def colormap(self):
+        """Ascii Palette derived from 'self.plugin_entry'.
+
+        Returns
+        -------
+        cmap: Matplotlob.colors.ListedColormap
+            - The colormap derived from the ascii palette
+        """
+        if not hasattr(self, "_colormap"):
+            path_to_ascii = str(
+                files(self.plugin_entry["package"]) / self.plugin_entry["relpath"]
+            )
+            self._colormap = from_ascii(path_to_ascii)
+        return self._colormap
+
+    def __call__(self, **kwargs):
+        """Generate a dictionary of colormap information used for linear norm cmaps.
+
+        This information used in both colorbar and image production throughout
+        GeoIPS image output specifications.
+
+        Parameters
+        ----------
+        kwargs: dict of arguments
+            - For a full list of arguments that will be provided to this function, see:
+              geoips.plugins.colormappers.ascii_based:call
+
+        Returns
+        -------
+        mpl_colors_info : dict
+            * Specifies matplotlib Colors parameters for use in both plotting and
+              colorbar generation
+
+        See Also
+        --------
+        :ref:`api`
+            See geoips.image_utils.mpl_utils.create_colorbar for field descriptions.
+        """
+        min_val = None
+        max_val = None
+        kwargs = SimpleNamespace(**kwargs)
+        if kwargs.data_range is not None:
+            min_val = kwargs.data_range[0]
+            max_val = kwargs.data_range[1]
+
+        mpl_cmap = self.colormap
+
+        LOG.info("Setting norm")
+
+        mpl_norm = colors.Normalize(vmin=min_val, vmax=max_val)
+        if kwargs.cbar_ticks:
+            mpl_ticks = kwargs.cbar_ticks
+        elif min_val is not None:
+            mpl_ticks = [int(min_val), int(max_val)]
+        else:
+            mpl_ticks = None
+
+        if kwargs.cbar_tick_labels is None:
+            mpl_tick_labels = mpl_ticks
+
+        # Must be uniform or proportional, None not valid for Python 3
+        mpl_boundaries = None
+
+        mpl_colors_info = {
+            "cmap": mpl_cmap,
+            "norm": mpl_norm,
+            "boundaries": mpl_boundaries,
+            "colorbar": kwargs.create_colorbar,
+            "cbar_ticks": mpl_ticks,
+            "cbar_tick_labels": mpl_tick_labels,
+            "cbar_label": kwargs.cbar_label,
+            "cbar_spacing": kwargs.cbar_spacing,
+            "cbar_full_width": kwargs.cbar_full_width,
+            "colorbar_kwargs": kwargs.colorbar_kwargs,
+            "set_ticks_kwargs": kwargs.set_ticks_kwargs,
+            "set_label_kwargs": kwargs.set_label_kwargs,
+        }
+
+        return mpl_colors_info
+
+
+class AsciiPaletteInterface(BaseTextInterface):
+    """Interface for the Ascii Palette (kid of Colormapper) to apply to the product."""
+
+    name = "ascii_palettes"
+    required_args = {"standard": {}}
+    required_kwargs = {"standard": {}}
+
+    @property
```

Comment:
Talk with Jeremy to see if there is another way to require an attribute from an Abstract Class

Reply:
See [here](https://github.com/NRLMMD-GEOIPS/geoips/blob/d47daffc69202cfb4c1f621fe238547871bc1f70/geoips/interfaces/base.py#L988) for the abstract implementation I want to address.

Reply:
Got an answer. Resolving.